### PR TITLE
Enable on demand GPU scheduling support for GKE

### PR DESCRIPTION
**Pull request by jpurusho65**
## Describe your changes and why you are making these changes

1. Add new resource providers
2. Add additional oauth scope to fetch nvidia drivers
3. Add the latest nvidia COS driver plugin terraform script

## Related issue number (if any)
ENG-3066

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x ] I have performed a self-review of my code.
- [x ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [x ] I have run the integration tests locally and they are passing.
- [x ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Comment by jpurusho65** [comment]
Screen Shot 2023-06-04 at 15 27 42 PM
Screen Shot 2023-06-04 at 15 33 23 PM

File: src/terraform/gke/variables.tf [hunk 1]
```diff
@@ -53,11 +53,13 @@ variable "max_cpu_node" {
 variable "min_gpu_node" {
   description = "Minimum number of nodes in the GPU node group"
   type        = number
+  default     = 1
```
Comment (cw75) [inline src/terraform/gke/variables.tf:56]: reminder to remove this default.

Comment (jpurusho65) [reply src/terraform/gke/variables.tf]: > reminder to remove this default.

done

File: src/terraform/gke/variables.tf [hunk 2]
```diff
@@ -53,11 +53,13 @@ variable "max_cpu_node" {
...
 }
 
 variable "max_gpu_node" {
   description = "Maximum number of nodes in the GPU node group"
   type        = number
+  default     = 2
```
[the remark this answers is not in the document]
Comment (jpurusho65) [reply src/terraform/gke/variables.tf]: done

**Review by cw75 — APPROVED** [review]
LGTM!

**Pull request merged**